pennylane/qchem/openfermion_obs.py

```diff
@@ -944,24 +944,22 @@ def molecular_hamiltonian(
         )
         if args is None:
             h = qml.qchem.diff_hamiltonian(mol, core=core, active=active)()
-            if grouping_type:
-                h = qml.Hamiltonian(
-                    qml.numpy.real(h.coeffs, requires_grad=False),
-                    h.ops,
-                    grouping_type=grouping_type,
-                    method=grouping_method,
-                )
-            if wires:
-                h = qml.map_wires(h, wires_map)
-            return h, 2 * len(active)
-        h = qml.qchem.diff_hamiltonian(mol, core=core, active=active)(*args)
-        if grouping_type:
             h = qml.Hamiltonian(
                 qml.numpy.real(h.coeffs, requires_grad=False),
                 h.ops,
                 grouping_type=grouping_type,
                 method=grouping_method,
             )
+            if wires:
+                h = qml.map_wires(h, wires_map)
+            return h, 2 * len(active)
+        h = qml.qchem.diff_hamiltonian(mol, core=core, active=active)(*args)
```

Comment:
The code looks somewhat repetitive here. Why don't we use `if-else` just for `qml.qchem.diff_hamiltonian(mol, core=core, active=active)(*args)` since the rest of the other things remain the same both inside the if and outside it?

```
        if args is None:
            h = qml.qchem.diff_hamiltonian(mol, core=core, active=active)()
        else:
            h = qml.qchem.diff_hamiltonian(mol, core=core, active=active)(*args)
        
        h = qml.Hamiltonian(
            qml.numpy.real(h.coeffs, requires_grad=False),
            h.ops,
            grouping_type=grouping_type,
            method=grouping_method,
        )
        if wires:
            h = qml.map_wires(h, wires_map)
        return h, 2 * len(active)
```

Reply:
Repeated lines removed.